js/zb.js

```diff
@@ -334,37 +334,115 @@ module.exports = class zb extends Exchange {
+        //             id: '100',
+        //             canCancelOrder: true,
+        //             area: '1',
+        //             mixMarginCoinName: 'usdt',
+        //             fundingRateRatio: '0.25',
+        //             marginCurrencyName: 'usdt',
+        //             minTradeMoney: '0.0001',
+        //             enableTime: '1638954000000',
+        //             maxTradeMoney: '10000000',
+        //             canTrade: true,
+        //             maxLeverage: '125',
+        //             defaultPositionsMode: '2',
+        //             onlyWhitelistVisible: false,
+        //             riskWarnRatio: '0.8',
+        //             marginDecimal: '8',
+        //             spot: false,
+        //             status: '1',
+        //             amountDecimal: '3',
+        //             leverage: false,
+        //             minAmount: '0.001',
+        //             canOrder: true,
+        //             duration: '1',
+        //             feeDecimal: '8',
+        //             sellerCurrencyId: '1',
+        //             maxAmount: '1000',
+        //             canOpenPosition: true,
+        //             isSupportMixMargin: false,
+        //             markPriceLimitRate: '0.05',
+        //             marginCurrencyId: '6',
+        //             stopFundingFee: false,
+        //             priceDecimal: '2',
+        //             lightenUpFeeRate: '0',
+        //             futures: true,
+        //             sellerCurrencyName: 'btc',
+        //             marketPriceLimitRate: '0.05',
+        //             canRebate: true,
+        //             marketName: 'BTC_USDT',
+        //             depth: [ 0.01, 0.1, 1 ],
+        //             createTime: '1607590430094',
+        //             mixMarginCoinIds: [ 6 ],
+        //             buyerCurrencyName: 'usdt',
+        //             stopService: false
+        //         },
+        //     }
+        //
+        const contractsData = this.safeValue (contracts, 'data', []);
+        const contractsById = this.indexBy (contractsData, 'marketName');
+        const dataById = this.deepExtend (contractsById, markets);
+        const keys = Object.keys (dataById);
         const result = [];
         for (let i = 0; i < keys.length; i++) {
             const id = keys[i];
-            const market = markets[id];
+            const market = dataById[id];
             const [ baseId, quoteId ] = id.split ('_');
             const base = this.safeCurrencyCode (baseId);
             const quote = this.safeCurrencyCode (quoteId);
-            const symbol = base + '/' + quote;
-            const amountPrecisionString = this.safeString (market, 'amountScale');
-            const pricePrecisionString = this.safeString (market, 'priceScale');
-            const priceLimit = this.parsePrecision (pricePrecisionString);
+            const settleId = this.safeValue (market, 'buyerCurrencyName');
```

Comment:
I feel like this variable might be `marginCurrencyName`, but I could be wrong.

I feel like `buyerCurrencyName` is the quote, and `sellerCurrencyName` is the base, but I could be wrong about this too

Reply:
I can agree with this I was thinking this as well